pkg/auth/providers/azuread/azuread_provider.go

```diff
@@ -0,0 +1,294 @@
+package azuread
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/mitchellh/mapstructure"
+	"github.com/pkg/errors"
+	"github.com/rancher/norman/types"
+	"github.com/rancher/rancher/pkg/auth/providers/common"
+	"github.com/rancher/types/apis/management.cattle.io/v3"
+	"github.com/rancher/types/config"
+	"github.com/rancher/types/user"
+
+	"github.com/rancher/norman/httperror"
+	"github.com/rancher/types/apis/management.cattle.io/v3public"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+const (
+	Name      = "azuread"
+	userType  = "user"
+	groupType = "group"
+)
+
+type azureADProvider struct {
+	ctx           context.Context
+	authConfigs   v3.AuthConfigInterface
+	azureADClient *Client
+	userMGR       user.Manager
+}
+
+func Configure(ctx context.Context, mgmtCtx *config.ScaledContext, userMGR user.Manager) common.AuthProvider {
+	azureADClient := &Client{
+		httpClient: &http.Client{},
```

Comment:
In 1.6, did we configure any timeouts with our client and/or http requests for azuerAD, if so, can you set timeouts here accordingly?

Reply:
I cannot find in timeout when configure settings in 1.6